### PR TITLE
Group unsupported object type messages

### PR DESCRIPTION
**Pull request by yellappam**
When number of unsupported objects increase in a package, user needs to scroll through the long list of messages. This change groups messages per object type reducing the number of messages. Also, the filtering before serialization is good in terms of effective work process utilization( in parallel mode ). Related issue [#4565](https://github.com/abapGit/abapGit/issues/4565)

Unit testing:
image

### CODE REVIEW

**Comment by yellappam** [comment]
Not sure, why abaplint is being checked on files which are not modified in the current PR. Also, some checks are not clear why they are reported. Example below
image

I am new to abaplint usage and might be missing something. Please suggest.

**Comment by larshp** [comment]
fix `IF NOT line_exists( lt_supported_types[ table_line = <ls_tadir>-object ] ).` first, its not valid 702 syntax

**Comment by yellappam** [comment]
Surprised to see other issues gone by fixing one issue. Curious to know what happened before.

**Comment by larshp** [comment]
errors in abaplint does cascade too much

**Review by mbtools — COMMENTED** [review]
some comments

File: src/objects/core/zcl_abapgit_serialize.clas.abap [hunk 1]
```diff
@@ -326,6 +334,40 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
...
 
+  METHOD filter_unsupported_objects.
+
+    DATA: ls_unsupported_count TYPE ty_unsupported_count,
+          lt_supported_types   TYPE zcl_abapgit_objects=>ty_types_tt,
+          lt_unsupported_count TYPE ty_unsupported_count_tt.
+
+    FIELD-SYMBOLS: <ls_tadir> LIKE LINE OF ct_tadir.
+
+    lt_supported_types = zcl_abapgit_objects=>supported_list( ).
```
[the remark this answers is not in the document]
Comment (yellappam) [reply src/objects/core/zcl_abapgit_serialize.clas.abap:345]: Supported object types cached

File: src/objects/core/zcl_abapgit_serialize.clas.abap [hunk 2]
```diff
@@ -326,6 +334,40 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
...
+      ENDIF.
+    ENDLOOP.
+    IF lt_unsupported_count IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    DELETE ct_tadir WHERE object IS INITIAL.
+    IF mi_log IS BOUND.
+      LOOP AT lt_unsupported_count INTO ls_unsupported_count.
+        mi_log->add_error( iv_msg  = |Object type { ls_unsupported_count-obj_type } not supported, {
```
[the remark this answers is not in the document]
Comment (yellappam) [reply src/objects/core/zcl_abapgit_serialize.clas.abap]: IS_ITEM is not visible in the log banner. If there is only one object name in the unsupported object type, object name is shown in the message. Output after changes

image

File: src/objects/core/zcl_abapgit_serialize.clas.abap [hunk 3]
```diff
@@ -326,6 +335,56 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
...
+
+    DELETE ct_tadir WHERE object IS INITIAL.
+    IF mi_log IS BOUND.
+      LOOP AT lt_unsupported_count ASSIGNING <ls_unsupported_count>.
+        IF <ls_unsupported_count>-count = 1.
+          mi_log->add_error( iv_msg  = |Object type { <ls_unsupported_count>-obj_type } not supported, {
+                                       <ls_unsupported_count>-obj_name } ignored| ).
+        ELSE.
+          mi_log->add_error( iv_msg  = |Object type { <ls_unsupported_count>-obj_type } not supported, {
+                                       <ls_unsupported_count>-count } object(s) ignored| ).
```
Comment (mbtools) [inline src/objects/core/zcl_abapgit_serialize.clas.abap:380]: change "object(s)" to "objects" since now it's always > 1 ;-)

**Review by mbtools — APPROVED** [review]
LGTM 👍

File: src/objects/core/zcl_abapgit_serialize.clas.abap (hunk 3 above)
Comment (yellappam) [reply src/objects/core/zcl_abapgit_serialize.clas.abap]: changes made

**Review by mbtools — APPROVED** [review]
lgtm 👍

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**